src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -56,14 +60,15 @@ public enum WorkItemQueueState
     /// </summary>
     public class WorkItemQueue
     {
+        private const int spinCount = 5;
+
         private Logger log = InternalTrace.GetLogger("WorkItemQueue");
 
-        private Queue<WorkItem> _innerQueue = new Queue<WorkItem>();
-        private object _syncRoot = new object();
-#if NETCF
-        private ManualResetEvent _syncEvent = new ManualResetEvent(false);
-        private int _waitCount = 0;
-#endif
+        private readonly ConcurrentQueue<WorkItem> _innerQueue = new ConcurrentQueue<WorkItem>();
+        private readonly ManualResetEventSlim _mreAdd = new ManualResetEventSlim(false);
+
+        private int _addId = int.MinValue;
+        private int _removeId = int.MinValue;
 
```

Comment:
Would you add a comment somewhere explaining the purpose of these ids in uniquely identifying each operation?